### PR TITLE
[HttpKernel] render_esi should behave in dev as it does in prod

### PR DESCRIPTION
**Pull request by bpaulin**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR | todo |
- [ ] gather feedback
- [ ] submit changes to the documentation

Hello,

If an object can't be turned into a string for the URL, then calling render_esi would throw an exception. 
But in dev env, render_esi silently falls back on using a InlineFragmentRenderer and everything seems to work.

I add a parameter (_framework.fragments.strict_) to make the fallback throw the same exception, to detect the logic error in dev mode, before deploying on an env with cache enabled.

### CODE REVIEW

**Comment by jevillard** [comment]
+1 for this feature @bpaulin

**Comment by fabpot** [comment]
Adding an option only makes things more difficult for no real benefit. Either we think that the current behavior is best, and no need to do anything, or we think the new behavior is much better and it should be the default.

**Comment by bpaulin** [comment]
I agree and I think the new behaviour would be better.
The option was here to avoid breaking compatibility.

I can rebase this pr and set this as default if needed

Thanks!
 Le 21 juin 2015 22:19, "Fabien Potencier" <notifications@github.com> a
écrit :

**Comment by bpaulin** [comment]
Hello again,
no more parameter now.
If InlineFragmentRenderer is called as fallback for EsiFragmentRenderer, it now checks the logic error.

@fabpot : I'm not sure if this pr should be on 2.7 now, there is a BC break after all.
everyone : any feedback?

File: src/Symfony/Component/HttpKernel/Fragment/InlineFragmentRenderer.php [hunk 1]
```diff
@@ -59,6 +59,11 @@ public function render($uri, Request $request, array $options = array())
             // want that as we want to preserve objects (so we manually set Request attributes
             // below instead)
             $attributes = $reference->attributes;
+            if (isset($options['is_fallback']) && $options['is_fallback'] )
+            {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Fragment/InlineFragmentRenderer.php]: Not sure why fabbot did not catch it, but we put opening braces on the same line as the `if`

File: src/Symfony/Component/HttpKernel/Tests/Fragment/EsiFragmentRendererTest.php [hunk 2]
```diff
@@ -90,12 +90,19 @@ public function testRenderAltControllerReferenceWithoutSignerThrowsException()
...
     {
         $inline = $this->getMockBuilder('Symfony\Component\HttpKernel\Fragment\InlineFragmentRenderer')->disableOriginalConstructor()->getMock();
 
         if ($called) {
-            $inline->expects($this->once())->method('render');
+            if ($options)
+            {
+                $inline->expects($this->once())->method('render')->with($this->anything(), $this->anything(), $options);
+            }
+            else
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Tests/Fragment/EsiFragmentRendererTest.php]: same here for if/else: one line

**Comment by jakzal** [comment]
> If an object can't be turned into a string for the URL, then calling render_esi would throw an exception. 
But in dev env, render_esi silently falls back on using a InlineFragmentRenderer and everything seems to work.

I suggest we treat it as a bug and consistently throw and exception in all environments (or at least, fallback to inline renderer consistently). We don't need an option here. There's also no BC break since no one could possibly rely on this inconsistency. If a bug like this slipped through dev, it would be discovered on production, investigated and fixed.

**Comment by xabbuh** [comment]
I agree. This should be consistent regardless the environment.

**Comment by fabpot** [comment]
closing in favor of #16283

**Pull request closed without merging**